sparkler-app/src/main/scala/edu/usc/irds/sparkler/pipeline/Crawler.scala

```diff
@@ -96,6 +102,14 @@ class Crawler extends CliTool {
       conf.setMaster(sparkMaster)
     }
     sc = new SparkContext(conf)
+
+    if(!path.isEmpty && path == "true"){
+      sc.addJar(getClass.getProtectionDomain.getCodeSource.getLocation.getPath)
+    }
+    else if(!path.isEmpty) {
+      sc.addJar(path)
```

Comment:
Instead of adding jar here, let's use SparkConf which accepts a list of jars at once. Please refer this - https://spark.apache.org/docs/1.6.1/api/java/org/apache/spark/SparkConf.html#setJars(java.lang.String[])